ql/src/java/org/apache/hadoop/hive/ql/parse/QBParseInfo.java

```diff
@@ -498,6 +504,21 @@ public void setExprToColumnAlias(ASTNode expr, String alias) {
     exprToColumnAlias.put(expr,  StringInternUtils.internIfNotNull(alias));
   }
 
+  public void setDestASTLimit(String dest, ASTNode limitExpr) {
+    destToASTLimit.put(dest, limitExpr);
+  }
+
+  public ASTNode getDestASTLimit(String dest) {
+    return destToASTLimit.get(dest);
+  }
+  public void setDestASTOffset(String dest, ASTNode offsetExpr) {
+    destToASTOffset.put(dest, offsetExpr);
+  }
+
+  public ASTNode getDestASTOffset(String dest) {
+    return destToASTOffset.get(dest);
+  }
+
```

Comment:
Instead of adding new methods and structures we could possibly make the existing ones more general; i.e., 
```
private final Map<String, SimpleEntry<Integer, Integer>> destToLimit;
private final Map<String, SimpleEntry<ASTNode, ASTNode>> destToLimit;
```